### ISSUE TITLE
Accessibility feature: Change text size

### ISSUE THREAD

**Issue by joenepraat** [mastodon/mastodon-android#569]
Please make it able to system independently change the text size in the timelines and in individual posts. Text size doesn't always need to be same (mostly larger) as the system wide text size.

**Comment by grishka** [comment]
Text size in posts, and in some other places where it won't break the layout, is already in `sp` units which means it scales with the system-wide text scaling factor.

**Comment by joenepraat** [comment]
> Text size in posts, and in some other places where it won't break the layout, is already in `sp` units which means it scales with the system-wide text scaling factor.

This issue is about independently from the system-wide text size.

**Comment by grishka** [comment]
So what?

**Comment by joenepraat** [comment]
> I'm very much against adding anything in-app for accessibility. If someone needs accessibility features, they usually enable them on the system level.

Yes, but that doesn't work for all people. That's why apps like Fedilab, Tusky and I thought even the Twitter app, gives the user the option to override system-wide text size settings.

**Comment by shiide** [comment]
As an additional anecdata-point: with a system text size that works for me, I find the app's timeline font to be too small to read comfortably.

I'd love a toggle for larger fonts, or a three-step text size slider. (Not the UX-challenged fiddling with a precision slider á la FediLab.)

Started to think about implementing this (reading the above though I'm not sure it'd be welcome). Might do it anyway as the list of usable Android Mastodon clients is short.

**Comment by fumin** [comment]
@grishka I have enabled the largest font on my android phone, and Mastodon's font are too small to read!!!!! Twitter is so much better than your app in this respect, why would you let Twitter get ahead of you so much!?

**Comment by joenepraat** [comment]
@fumin Please have some respect. Thanks.

**Comment by Matth7878** [comment]
It's good text scale with system. Yet I want to advocate for this feature : depending on what you're looking at you may use a bigger or smaller text size.

For instance I don't want to have small text for my launcher, mail, messages and so on... For Mastodon when I am looking at it I want to first be able to quickly view who posted new things and only then eventually read what they said. So density and how many posts it's possible to see at once is important and would require smaller text. 
To accommodate Mastodon it would force me to choose between what I am comfortable for my home screen, mails, messages and specifically Mastodon.

Aa previous users pointed out I believe it's something that is widely known as a lot of apps let users change it. Twitter, WhatsApp, even Chrome and Google Message allow you to zoom which is surprising for a Google apps to let you do it.

**Comment by Revertron** [comment]
The best way to set font size (as an example) is in K-9 Mail. They allow to set the size in percents of normal sizes.
